forte/proc/aset2.py

```diff
@@ -0,0 +1,309 @@
+# @BEGIN LICENSE
+#
+# forte_inversion by Psi4 Developer, a plugin to:
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2016 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+import psi4
+import forte
+import numpy as np
+from forte.proc.dsrg import ProcedureDSRG
+
+def aset2_driver(state_weights_map, scf_info, ref_wfn, mo_space_info, options):
+    """
+    Driver to run an systematic Active Space Embedding Theory (ASET) computation.
+
+    :param state_weights_map: dictionary of {state: weights}
+    :param scf_info: a SCFInfo object of Forte
+    :param ref_wfn: a Psi4::Wavefunction object
+    :param mo_space_info: a MOSpaceInfo object with embedding partitions (EMBEDDING_ORB)
+    :param options: a ForteOptions object of Forte
+
+    :return: the computed ASET energy.
+    """
+
+    # Read all options
+    frag_as_solver = options.get_str('ACTIVE_SPACE_SOLVER')
+    frag_do_fci = options.get_bool('FRAG_DO_FCI')
+    frag_corr_solver = options.get_str('FRAG_CORRELATION_SOLVER')
+    frag_corr_level = options.get_str('FRAG_CORR_LEVEL')
+    env_corr_solver = options.get_str('ENV_CORRELATION_SOLVER')
+    env_corr_level = options.get_str('ENV_CORR_LEVEL')
+    relax = options.get_str('RELAX_REF')
+    semi = options.get_bool('SEMI_CANONICAL')
+    int_type_frag = options.get_str('INT_TYPE_FRAG')
+    int_type_env = options.get_str('INT_TYPE_ENV')
+    do_aset_mf = options.get_bool('EMBEDDING_ASET2_MF_REF')
+    frag_density = options.get_str('FRAGMENT_DENSITY')
+    ref_type = options.get_str('EMBEDDING_REFERENCE')
+    name = frag_corr_solver + '-' + frag_corr_level
+    #fold = options.get_bool('DSRG_FOLD')
+    
+    # Build two option lists for fragment (A) and environment (B) solver
+    A_list = [frag_corr_solver, frag_corr_level, int_type_frag, relax, semi]
+    B_list = [env_corr_solver, env_corr_level, int_type_env]
+
+    psi4.core.print_out("\n  ")
+    psi4.core.print_out("\n  ")
+    forte.print_method_banner(["Second-order Active Space Embedding Theory [ASET(2)]", "Nan He, Chenyang Li"])
```

Comment:
I think you can safely put only your name here :)

Reply:
Thanks!